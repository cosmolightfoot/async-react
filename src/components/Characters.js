import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';

function Characters({ characters }) {
  const charList = characters.map((char, i) => {
    return <Character key={i} character={char} />;
  });

  return (
    <ul>
      {charList}
    </ul>
  );
}

Characters.propTypes = {
  characters: PropTypes.array.isRequired
};

export default Characters;
