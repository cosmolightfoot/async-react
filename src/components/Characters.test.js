import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters';

describe('characters tests', () => {
  it('snapshot test', () => {
    const wrapper = shallow(<Characters characters={[{ is:'array' }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
