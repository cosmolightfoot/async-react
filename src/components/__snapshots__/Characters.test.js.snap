// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`characters tests snapshot test 1`] = `
<ul>
  <Character
    character={
      Object {
        "is": "array",
      }
    }
    key="14fac065-68da-463d-9689-381d706ec11e"
  />
</ul>
`;
